core/src/Kokkos_HostSpace.hpp

```diff
@@ -291,15 +293,17 @@ namespace Kokkos {
 
 namespace Impl {
 
+#define PAR_DEEP_COPY_USE_MEMCPY
+
 template< class ExecutionSpace >
 struct DeepCopy< HostSpace, HostSpace, ExecutionSpace > {
   DeepCopy( void * dst, const void * src, size_t n ) {
-    memcpy( dst, src, n );
+    hostspace_parallel_deepcopy(dst,src,n);
```

Comment:
This needs to probably forward the execution space argument. You can then specialize it for OpenMP / Serial if you want to, and or use proper parallel_for from the execution space.

Reply:
Using SFINAE to avoid some of the branches would be ideal, and prevent the issue you listed below (OpenMP used when it wasn't requested).  I don't think I can access an 'ExecutionSpace' 'type' here, because to include the headers creates a circular dependency.

I can still use SFINAE to do the right thing though (I think).

Reply:
SFINAE did not seem necessary, a simply specialization works.

There is a problem though, the ExecutionSpace is not a concrete type, so calling ExecSpace.fence() won't work.  Perhaps my Metaprogramming Foo is weak, but the only way I see that working is by including the full definition of the Execution space... which IMO is undesired.  If there is another way to call Kokkos::OpenMP::fence, please explain how to do it.